src/comments/comments.jsx

```diff
@@ -0,0 +1,176 @@
+import React, { useRef, useState, useEffect } from 'react';
+import './comments.css'
+import { useDispatch, useSelector } from 'react-redux';
+import { allCommentsList } from '@/store/selectors';
+import moment from 'moment';
+import { changeCommentText, changeQuantityLikes, deleteComment, addComment } from '@/features/allComments/allComments';
+import ReactModal from 'react-modal';
+import Delete from '@/icons/delete';
+import Edit from '@/icons/edit';
+import { currentUser } from '@/store/selectors';
+import { v4 } from 'uuid';
+const dateDifference = (startDate) => {
+    const endDate = moment().format('YYYY-MM-DD HH:mm')
+    const startMoment = moment(startDate, 'YYYY-MM-DD HH:mm');
+    const endMoment = moment(endDate, 'YYYY-MM-DD HH:mm');
+    const diffDuration = moment.duration(endMoment.diff(startMoment));
+    const days = diffDuration.asDays();
+    const hours = diffDuration.asHours();
+    const minutes = diffDuration.asMinutes();
+
+    if (days >= 1) {
+        return Math.floor(days) + ' days ago';
+    } else if (hours >= 1) {
+        return Math.floor(hours) + ' hours ago';
+    } else {
+        return Math.floor(minutes) + ' minutes ago';
+    }
```

Comment:
no need to do this all manually, moment.js has [fromNow](https://momentjs.com/docs/#/displaying/from/) helper. reading the docs of the library that you use usually helps